### PR TITLE
Add optional config when embedding voyager as library

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Add default embed configuration to `VoyagerConfig`.
- Currently it supports hiding header and footer.
- This configuration basically gives an option to not render the header and footer when embedding voyager in another application.
- By default the header and footer are rendered. So existing setup requires no change.

Created this as part of the discussion that went in this Issue: https://github.com/vega/voyager/issues/673

#### Note:
- Have not modified any docs. Not sure if this is the final implementation or if there is a simpler way to do this.

### CODE REVIEW

**Comment by ajainarayanan** [comment]
@kanitw As discussed in #673 I have added an optional configuration to show/hide header and footer. Let me know if I have missed anything.

File: config/webpack.config.dev.js [hunk 1]
```diff
@@ -132,9 +132,9 @@ module.exports = {
   // assume a corresponding global variable exists and use that instead.
   // This is important because it allows us to avoid bundling all of our
   // dependencies, which allows browsers to cache those libraries between builds.
+
+  // Since its a local dev build we don't need to mark anything external.
```
Comment (kanitw) [inline config/webpack.config.dev.js]: Like commented above, not bundling all of our dependencies allow browsers to cache those libraries and may make the bundling faster?

File: src/components/app-root/index.tsx [hunk 2]
```diff
@@ -39,13 +45,15 @@ class AppRootBase extends React.PureComponent<AppRootProps, {}> {
...
+        if (!hideFooter) {
+          footer = <Footer/>;
+        }
       }
     }
     return (
       <div className="voyager">
         <LogPane/>
-        <Header/>
+        {!hideHeader ? <Header/> : null}
```
Comment (kanitw) [inline src/components/app-root/index.tsx]: `!hideHeader && <Header/>`

File: src/lib-voyager.tsx [hunk 3]
```diff
@@ -44,7 +44,7 @@ export class Voyager {
       this.container = container;
     }
 
-    this.config = config;
+    this.config = Object.assign({}, DEFAULT_VOYAGER_CONFIG, config);
```
Comment (kanitw) [inline src/lib-voyager.tsx]: `{...DEFAULT_VOYAGER_CONFIG, ...config}`

(We use the spread operator in this project.)

File: src/models/config.ts [hunk 4]
```diff
@@ -1,11 +1,20 @@
+export interface EmbedProps {
+  hideHeader?: boolean;
+  hideFooter?: boolean;
+};
 export interface VoyagerConfig {
   showDataSourceSelector?: boolean;
   serverUrl?: string | null;
   manualSpecificationOnly?: boolean;
+  embedProps?: EmbedProps;
```
Comment (kanitw) [inline src/models/config.ts]: Let's not include unnecessary nesting.  Most of props are for using Voyager as a library anyway.  

Just make `hideHeader/Footer` properties of VoyagerConfig.

File: config/webpack.config.dev.js (hunk 1 above)
Comment (ajainarayanan) [reply config/webpack.config.dev.js]: Sorry didn't mean to overwrite. Since we modified external thought this might be an issue. Have reverted back.

**Comment by ajainarayanan** [comment]
@kanitw Have updated the PR.

File: src/components/app-root/index.tsx [hunk 5]
```diff
@@ -20,11 +22,13 @@ import {ViewPane} from '../view-pane/index';
...
 export interface AppRootProps {
   dataset: Dataset;
+  config: VoyagerConfig;
 }
 
 class AppRootBase extends React.PureComponent<AppRootProps, {}> {
   public render() {
-    const {dataset} = this.props;
+    const {dataset, config} = this.props;
+    const {hideHeader = false, hideFooter = false} = config;
```
Comment (kanitw) [inline src/components/app-root/index.tsx]: Given that we already set default somewhere else, maybe the default parameter here is unnecessary?

Comment (ajainarayanan) [reply src/components/app-root/index.tsx]: Its just as an extra precaution. I guess I really need to get used to typescript :) If its un-necessary I can remove this.

**Comment by ajainarayanan** [comment]
@kanitw Have removed the defaults as they are provided by the config.

**Comment by ajainarayanan** [comment]
@kanitw Thanks! Are there any tentative time line when you would cut a release

**Comment by kanitw** [comment]
We plan to cut another release sometimes this week.

**Pull request merged**